invenio_rdm_records/services/communities/service.py

```diff
@@ -41,19 +41,23 @@ def create(self, identity, data, record, uow=None):
         """Include the record in the given communities."""
         pass
 
-    def _remove(self, community_id, record):
+    def _remove(self, identity, community_id, record):
         """Remove a community from the record."""
         if community_id not in record.parent.communities.ids:
             raise RecordCommunityMissing(record.id, community_id)
 
+        # check permission here, per community: curator cannot remove another community
+        self.require_permission(
+            identity, "remove_community", record=record, community_id=community_id
+        )
+
         # Default community is deleted when the exact same community is removed from the record
         record.parent.communities.remove(community_id)
 
     @unit_of_work()
-    def delete(self, identity, record_id, data, revision_id=None, uow=None):
+    def remove(self, identity, record_id, data, revision_id=None, uow=None):
```

Comment:
we might need to have a `delete` overwritten and `NotImplemented(use remove instead)`. Otherwise programatically we might reach https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/records/service.py#L468 and have unknwon/undesired effect.

Reply:
Here we have changed approach and we completely create a new service without inheriting from `RecordService`. We inherit from the [base class](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/base/service.py#L15).
IMO, when create a service to CRUD records, than we should stick to that interface. For custom ad-hoc services, I would instead create APIs with more meaningful names then REST APIs verbs: in this case `add` and `remove`, instead of `update` and `delete`.

Reply:
LGTM, I missed the inheritance change